client/blocks/comment-detail/index.jsx

```diff
@@ -188,32 +184,27 @@ export class CommentDetail extends Component {
 	}
 }
 
-const mapStateToProps = ( state, { commentId, siteId } ) => {
-	//const comment = getComment( state, siteId, commentId );
-	const comment = mockComment;
-
-	return {
-		authorAvatarUrl: get( comment, 'author.avatar_URL' ),
-		authorDisplayName: get( comment, 'author.name' ),
-		authorEmail: get( comment, 'author.email' ),
-		authorId: get( comment, 'author.ID' ),
-		authorIp: get( comment, 'author.ip' ), // TODO: not available in the current data structure
-		authorIsBlocked: get( comment, 'author.isBlocked' ), // TODO: not available in the current data structure
-		authorUrl: get( comment, 'author.URL' ),
-		authorUsername: get( comment, 'author.nice_name' ),
-		commentContent: comment.content,
-		commentDate: comment.date,
-		commentId: commentId,
-		commentIsApproved: 'approved' === comment.status,
-		commentIsLiked: comment.i_like,
-		commentIsSpam: 'spam' === comment.status,
-		commentIsTrash: 'trash' === comment.status,
-		postAuthorDisplayName: get( comment, 'post.author.name' ),
-		postTitle: get( comment, 'post.title' ),
-		postUrl: get( comment, 'post.link' ),
-		repliedToComment: comment.replied, // TODO: not available in the current data structure
-		siteId: siteId,
-	};
-};
+const mapStateToProps = ( state, ownProps ) => ( {
+	authorAvatarUrl: get( ownProps, 'author.avatar_URL' ),
+	authorDisplayName: get( ownProps, 'author.name' ),
+	authorEmail: get( ownProps, 'author.email' ),
+	authorId: get( ownProps, 'author.ID' ),
+	authorIp: get( ownProps, 'author.ip' ), // TODO: not available in the current data structure
+	authorIsBlocked: get( ownProps, 'author.isBlocked' ), // TODO: not available in the current data structure
+	authorUrl: get( ownProps, 'author.URL' ),
+	authorUsername: get( ownProps, 'author.nice_name' ),
+	commentContent: ownProps.content,
+	commentDate: ownProps.date,
+	commentId: ownProps.commentId,
+	commentIsApproved: 'approved' === ownProps.status,
+	commentIsLiked: ownProps.i_like,
+	commentIsSpam: 'spam' === ownProps.status,
+	commentIsTrash: 'trash' === ownProps.status,
+	postAuthorDisplayName: get( ownProps, 'post.author.name' ),
+	postTitle: get( ownProps, 'post.title' ),
+	postUrl: get( ownProps, 'post.link' ),
+	repliedToComment: ownProps.replied, // TODO: not available in the current data structure
+	siteId: ownProps.siteId,
+} );
```

Comment:
Something I hope to do soon is get a comment object defined somewhere and write a mapping function from this API call into that object. @rodrigoi I'll make sure you are part of it too. That way, in the end we should be able to safely use nesting of properties where appropriate. For example, it might eventually make more sense to use `comment.isSpam` or `const { isSpam, isLiked } = comment` instead of `commentIsSpam`. Further, I'd like to challenge us to think outside the box on some of these properties and dig deep into the essence of what a comment _is_. For example, a comment _cannot_ be both `spam` _and_ `trash`, so why does our data object allow that invalid representation?